### ISSUE TITLE
ST failures due to calicoctl emitting "ignoring ServerName" warning

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/calicoctl#1657]
This build failed due to failing to parse the YAML emitted by calicoctl: https://semaphoreci.com/calico/calicoctl/branches/pull-request-1656/builds/3

## Expected Behavior
YAML should be parsable.

## Current Behavior
It looks like a log got mixed into the output and then the test tried to parse it; frustratingly, there's very little context in the log:
```
======================================================================
ERROR: Run a mainline multi-host test [with test_type='rules.tags'].
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose_parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/code/tests/st/policy/test_profile.py", line 67, in test_multi_host
    self._setup_workloads(host1, host2)
  File "/code/tests/st/policy/test_profile.py", line 299, in _setup_workloads
    assert_number_endpoints(host1, 4)
  File "/code/tests/st/utils/utils.py", line 153, in wrapped
    return fn(*args, **kwargs)
  File "/code/tests/st/utils/utils.py", line 241, in assert_number_endpoints
    output = yaml.safe_load(out)
  File "/usr/lib/python2.7/site-packages/yaml/__init__.py", line 93, in safe_load
    return load(stream, SafeLoader)
  File "/usr/lib/python2.7/site-packages/yaml/__init__.py", line 71, in load
    return loader.get_single_data()
  File "/usr/lib/python2.7/site-packages/yaml/constructor.py", line 37, in get_single_data
    node = self.get_single_node()
  File "/usr/lib/python2.7/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/usr/lib/python2.7/site-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/usr/lib/python2.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python2.7/site-packages/yaml/composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "/usr/lib/python2.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/lib/python2.7/site-packages/yaml/parser.py", line 439, in parse_block_mapping_key
    "expected <block end>, but found %r" % token.id, token.start_mark)
ParserError: while parsing a block mapping
  in "<string>", line 1, column 1:
    2017-06-06 10:58:24.520844 I | w ... 
    ^
expected <block end>, but found '-'
  in "<string>", line 2, column 1:
    - apiVersion: v1
    ^
```

## Possible Solution
Prevent calicoctl from emitting logs by default.

Quite possible that the test harness is slurping up stderr when it should only slurp stdout.  However, we're not expecting any stderr output so maybe best not to fix the test harness.

## Steps to Reproduce (for bugs)
I've only seen it on 1/3 of ST runs in the PR above.

**Comment by fasaxc** [comment]
Ah, found some more context in the logs:
```
docker exec -it host1 sh -c 'export ETCD_AUTHORITY=etcd-authority-ssl:2379; export ETCD_SCHEME=https; export ETCD_CA_CERT_FILE=/home/runner/calicoctl/certs/ca.pem; export ETCD_CERT_FILE=/home/runner/calicoctl/certs/client.pem; export ETCD_KEY_FILE=/home/runner/calicoctl/certs/client-key.pem; /code/dist/calicoctl get workloadEndpoint -o yaml'
  # 2017-06-06 10:58:24.520844 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
  # - apiVersion: v1
  #   kind: workloadEndpoint
  #   metadata:
  #     name: 0c4b141121e31b2e04825dfaf05558f190b6608ca52287ee3022f7fa76798bf6
  #     node: 0da066fda1cd
  #     orchestrator: libnetwork
  #     workload: libnetwork
  #   spec:
```

**Comment by caseydavenport** [comment]
Hopefully fixed by https://github.com/projectcalico/calicoctl/pull/1656, which pins to a coreos/etcd v3.2 RC.

If that fixes it, we should keep this to track updating the pin once 3.2 is GA.

**Comment by caseydavenport** [comment]
Fixed by #1660.
